### DISCUSSION TITLE
Can we create our own modified node in distributed architecture ?

### DISCUSSION THREAD

**Discussion by Atindra30** [Security-Onion-Solutions/securityonion#9606] [Unsupported Versions] [answered]
I have to create node other than sensor and search node which can connect to master node, In how many files do we need to change?

**Comment by InfosecGoon** [comment, marked as answer]
I'm sorry, I'm not sure what you're asking. What are you trying to do with this new node, and what data does it need from the Manager?

**Reply by Atindra30** [reply]
Like we have workstation option is setup script, I want to create a new minion which can connect to manager  and after that i will add some components in that minion later according to my requirements. So I just wanted to known in which files(like top.sls, allowed_states.map.jinja) I have to add entries  of my new minion so that it can conect with manager.

**Reply by InfosecGoon** [reply]
So you want to use the Salt master process running on the manager to configure a new node type that you're designing?

**Reply by Atindra30** [reply]
yes

**Reply by Atindra30** [reply]
can we create our own node??

**Reply by InfosecGoon** [reply]
After speaking with the developers, this might be possible but we really don't recommend or support it. There are role checks throughout many states that would need to be modified, as well as the top and allowed_states configurations. Trying to retro-fit all of this would be significantly more work than running a separate Salt instance for your new node type, and would be very difficult to keep from breaking in future updates.
